README.md

```diff
@@ -6,12 +6,243 @@
 [![License](https://img.shields.io/cocoapods/l/Cache.svg?style=flat)](http://cocoadocs.org/docsets/Cache)
 [![Platform](https://img.shields.io/cocoapods/p/Cache.svg?style=flat)](http://cocoadocs.org/docsets/Cache)
 
+## Table of Contents
+
+* [Description](#description)
+* [Key features](#key-features)
+* [Usage](#usage)
+  * [Hybrid cache](#hybrid-cache)
+  * [Strict cache](#strict-cache)
+  * [Expiry date](#expiry-date)
+  * [Cachable protocol](#cachable-protocol)
+* [Optional bonuses](#optional-bonuses)
+  * [JSON](#json)
+  * [DefaultCacheConverter](#defaultcacheconverter)
+* [What about images?](#what-about-images)
+* [Installation](#installation)
+* [Author](#author)
+* [Contributing](#contributing)
+* [License](#license)
+
+## Description
+
+**Cache** doesn't claim to be unique in this area, but it's not another monster
+library that gives you a god's power.
+So don't ask it to fetch something from network or magically set an image from
+URL to your `UIImageView`.
+It does nothing but caching, but it does it well, having everything you expect
+from this kind of library. It offers a good public API with out-of-box
+implementations and great customization possibilities.   
+
+## Key features
+
+- Generic `Cachable` protocol to be able to cache any type you want.
+- `CacheAware` and `StorageAware` protocols to implement different kinds
+of key-value cache storages. The basic interface includes methods to add, get
+and remove objects by key.
+- `Cache` class to create a strict cache storage by a given name for a specified
+`Cachable`-compliant type.
+- `HybridCache` class that works with every kind of `Cachable`-compliant types.
+- Flexible `Config` struct which is used in the initialization of `Cache` and
+`HybridCache` classes, based on the concept of having front- and back- caches.
+A request to a front cache should be less time and memory consuming (`NSCache` is used
+by default here). On the other hand, back cache should be more permanent and
+independent from the life cycle of application, because it's more like backup
+solution to store your data (Disk cache is one of the reliable approaches here).
```

Comment:
> On the other hand, back cache should be more permanent and
+independent from the life cycle of application, because it's more like backup
+solution to store your data (Disk cache is one of the reliable approaches here).

> The difference between front and back caching is that back caching is used for content that outlives the application life-cycle. See it more like a convenient way to store user information that should persist across application launches. Disk cache is the most reliable choice here.